plugins/fs/src/commands.rs

```diff
@@ -26,10 +26,26 @@ use crate::{scope::Entry, Error, FilePath, FsExt};
 #[derive(Debug, serde::Deserialize)]
 #[serde(untagged)]
 pub enum SafeFilePath {
+    #[serde(deserialize_with = "deserialize_url")]
     Url(url::Url),
     Path(SafePathBuf),
 }
 
+fn deserialize_url<'de, D>(deserializer: D) -> Result<url::Url, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let url = url::Url::deserialize(deserializer)?;
+    let scheme = url.scheme();
+    if scheme.len() != 1 {
+        Ok(url)
+    } else {
+        Err(serde::de::Error::custom(format!(
+            "Single letter scheme \"{scheme}\" is not supported because it conflicts with Windows paths"
+        )))
+    }
```

Comment:
instead of an error here we should fallback to using a `SafePathBuf`, so instead we should implement `Deserialize` for `SafeFilePath` manually, not just one variant.